content/blog/intro-to-gamedev/index.md

```diff
@@ -0,0 +1,560 @@
+
+So let's go ahead and implement all this on our `GameApp` class.:
+
+```ts
+// creating an alias for our Union type that we can use
+type WorldObjects = Player | ScrollingObject
+
+export class GameApp {
+  public app: PIXI.Application
+
+  static PressedSpace: boolean = false
+  static Stage: PIXI.Container
+  static ActiveEntities: Array<WorldObjects> = new Array<WorldObjects>()
+  static GameOver: boolean = false
+  static ScrollSpeed: number = 3
+
+  static GroundPosition: number = 0 // ground position, given by screen height
+  static Width: number = 0 // width of game screen, given by screen width
+
+  // score of current run
+  static Score: number = 0
+
+  // max score achieved in session
+  static MaxScore: number = 0
+
+  // next score in which we should place an obstacle
+  static ScoreNextObstacle: number = 0
+}
+```
+
+Objects whose types are of this Union type (`Player | ScrollingObject`) have accessible members that are defined both on our Player and on ScrollingObject types. As seen above, I created a new static member called `ActiveEntities` which is an array of `WorldObject`s. We now have to change `GameApp.Update()` to loop over this array.
+
+But before that, you might have noticed as well that I created a couple more variables for tracking score stuff. This will allow us to do a couple of things:
+
+- Keep score for the current run
+- Save the maximum score achieved in a session
+- Know when the next obstacle should spawn (each time we create one, we set the score at which the next one should appear)
+
+To keep the code simple for now, everytime we spawn an obstacle we'll spawn a cloud for aesthetics. There's no real reason why we couldn't add a more random element to these decorations later, but for now this will keep the code easier to study/describe.
+
+Keeping that in mind, let's see what our current game update looks like:
+
+```ts
+    static Update(delta: number) {
+
+        // if we haven't lost yet let's update everything, otherwise wait for spacebar press to restart game
+        if (!this.GameOver)
+        {
+            // loop over object list
+            for(var i = 0; i < GameApp.ActiveEntities.length; i++)
+            {
+                // update entity
```

Comment:
the formatting is _really_ foreign to Typescript and Javascript so using prettier might make it a little bit closer to the heart of your audience